tensorflow_addons/metrics/r_square.py

```diff
@@ -86,13 +86,17 @@ def __init__(
         self,
         name: str = "r_square",
         dtype: AcceptableDTypes = None,
-        y_shape: Tuple[int, ...] = (),
         multioutput: str = "uniform_average",
         num_regressors: tf.int32 = 0,
         **kwargs,
     ):
         super().__init__(name=name, dtype=dtype, **kwargs)
-        self.y_shape = y_shape
+
+        if "y_shape" in kwargs:
+            warnings.warn(
+                "y_shape has been removed, because it's automatically derived.",
```

Comment:
Generally we indicate a release margin:

https://github.com/tensorflow/addons/blob/37a368adfea1d0ec4cb85998946f458e9d565818/tensorflow_addons/optimizers/lamb.py#L83

/cc @seanpmorgan Do you remember how much releases?